contracts/libraries/AchievementLib.sol

```diff
@@ -48,18 +48,34 @@ library AchievementLib {
   )
     internal
   {
-    if (achievementsType == AchievementCommonLib.AchievementsType.Rating) {
-      AchievementConfig storage achievementConfig;
-      for (uint64 i = 1; i <= achievementsContainer.achievementsCount; i++) { /// optimize ??
-        achievementConfig = achievementsContainer.achievementsConfigs[i];
+    AchievementConfig storage achievementConfig;
+    for (uint64 i = 1; i <= achievementsContainer.achievementsCount; i++) { /// optimize ??
+      achievementConfig = achievementsContainer.achievementsConfigs[i];
+      if (achievementsType != achievementConfig.achievementsType) continue;
 
-        if (achievementConfig.maxCount <= achievementConfig.factCount) continue;    // not exit/max
-        if (achievementConfig.lowerBound > currentValue) continue;
-        if (achievementsContainer.userAchievementsIssued[user][i]) continue; //already issued
-        achievementConfig.factCount++;
-        achievementsContainer.userAchievementsIssued[user][i] = true;
-        achievementsContainer.peeranhaNFT.mint(user, i);
-      }
+      if (achievementConfig.maxCount <= achievementConfig.factCount) continue;    // not exit/max
+      if (achievementConfig.lowerBound > currentValue) continue;
+      if (achievementsContainer.userAchievementsIssued[user][i]) continue; //already issued
+      achievementConfig.factCount++;
+      achievementsContainer.userAchievementsIssued[user][i] = true;
+      achievementsContainer.peeranhaNFT.mint(user, i);
     }
   }
+
+  function mintNFT(
```

Comment:
The name is not descriptive. It should explain that this method is only for manual NFTs. E.g. `mintManualNFT`